rln-research/membership-merkle-tree-storage-evaluation.md

```diff
@@ -0,0 +1,107 @@
+Below, we demonstrate the storage requirement to persist a group of size`1000`  under different Merkle tree depths: `d=[10..20]`. 
+
+d=`10` `96400` bytes
+d=`11` `96480` bytes
+d=`12` `96560` bytes
+d=`13` `96640` bytes
+d=`14` `96720` bytes
+d=`15` `96800` bytes
+d=`20` `97200` bytes
+
+As the results show, the storage overhead increases by an additive factor of `80 bytes` for every extra depth. Take `d=10` and `d=11` as an example, the difference in their storage overhead is `96480 - 96400 = 80` bytes. This trend holds for any other group size.  Thus, if the Merkle tree depth is set to `d=20`, and we end up having a group size of only `1024` members i.e., `2^10` members, then the extra storage overhead is `(20-10) *80 = 0.8 KB` bytes which can be considered minimal and insignificant.
+Note that this analysis is only valid for the storage overhead, however, setting a higher value for the depth may have a different impact on the computational load. This needs to be evaluated separately.
+
+
+
+### FMT vs OMT
+
+Comparing the max storage overhead of FMT and OMT, as depicted in the figure below, brings us to the conclusion that FMT is more efficient than OMT. However, this is only true for the worst-case scenario where the group is full to the maximum tree capacity.
+![Max relative](https://i.imgur.com/G8COahY.png)
+
+But, this worst-case scenario does not always hold, instead, in most states of the system, the group size is much less than the tree capacity. In the following figure, we take a snapshot of the system at group size `1000`. In this snapshot, we compare the storage overhead incurred by FMT and OMT for `d=[10..20]`.
+![group 1000](https://i.imgur.com/NuIDALV.png)
+
+As you can see, in this more realistic scenario, FMT shows better performance than OMT. For example, for `d=20`, persisting a group of `1000` members using OMT requires  `0.097 MB` whereas it requires `67.1 MB` when using FMT, `700` times more than OMT.
+
+It brings us to another conclusion that OMT outperforms FMT when group size is less than the tree capacity. However, the exact point at which this relationship changed needs more investigation.
+
+
+
+
+
+## Conclusion
+1) Storage overhead of membership MT does not seem to be an issue for desktop nodes given that the maximum overhead to persist an MT for a large group of `2^20` and `2^10` members does not exceed `100 MB` (see below for more accurate numbers).
+
+
+| Algorithm\Tree depth | d=10      | d=20      |
+| -------------------- | --------- | --------- |
+| FMT                  | 33.55  MB | 67.10 MB  |
+| OMT                  | 50.33  MB | 100.66 MB |
```

Comment:
I'm confused by this, is that saying 2^10 (=1024 members?) requires 50mb for OMT? That seems high to me, and odd if 1000 active leaves for 2^20 requires 0.097 MB. Are these numbers accurate?

Reply:
The values under `d=10` are indeed for `d=19`. Fixed them in https://github.com/vacp2p/research/pull/132/commits/f840bb918bafb142e2cfd7aeba3300edbeecd9cf
for d=10, OMT requires 0.098 MB , FMT needs 0.065 MB. Note that FMT always outperforms OMT when group size is equal to the tree capacity.